code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -131,92 +152,119 @@ fn stake_in_case_of_zero_inflation_should_work() {
 					duration: duration_preset,
 					unlock_penalty: rewards_pool.lock.unlock_penalty,
 				},
+				financial_nft_id: fnft_instance_id
 			})
 		);
+
 		assert_eq!(balance(staked_asset_id, &staker), amount);
-		assert_eq!(balance(staked_asset_id, &StakingRewards::pool_account_id(&pool_id)), amount);
-		assert_last_event::<Test, _>(|e| {
+		assert_eq!(balance(staked_asset_id, &fnft_asset_account), amount);
+
+		assert_last_event_with::<Test, _>(|event| {
 			matches!(
-				e.event,
-				Event::StakingRewards(crate::Event::Staked { pool_id, owner, amount, position_id, ..})
-					if owner == staker && pool_id == 1 && amount == 100_500u32.into() && position_id == 1
+				event,
+				Event::StakingRewards(crate::Event::Staked {
+					pool_id: PICA::ID,
+					owner,
+					amount: _,
+					duration_preset: _,
+					fnft_collection_id: PICA::ID,
+					fnft_instance_id: _,
+					keep_alive: _,
+				}) if owner == staker
 			)
+			.then_some(())
 		});
 	});
 }
 
+// this is almost the exact same as the above function
+// spot the difference!
+// maybe do a proptest with different inflation rates?
```

Comment:
Yes :) we could and we have a [task](https://app.clickup.com/t/2hr9vn9) for adding proptests for possible cases. But the idea is to iterate based on the existing functionality rather than try to build the best version(including tests) upfront which is more costly as we don't get feedback on how the product fits together.

Reply:
for sure - i just wanted to document it in the code for this specific case since it's such a good candidate for a proptest-like test